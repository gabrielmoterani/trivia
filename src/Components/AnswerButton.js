import React from 'react';
import { PropTypes } from 'prop-types';
import { StyleSheet } from 'react-native';
import { Button } from 'react-native-ui-kitten';

function AnswerButton(props) {
  const hasAnswer = props.answer !== undefined;
  const answerIsValue = props.answer === props.value;
  const isCorrect = props.answer && props.correct;
  const answerCallback = () => (
    !hasAnswer && props.callback && props.callback(props.value)
  );

  return (
    <Button
      size="large"
      appearance={answerIsValue ? 'filled' : 'outline'}
      status={answerIsValue ? (isCorrect ? 'success' : 'danger') : 'primary'}
      style={Style.answerBtn}
      onPress={answerCallback}
    >
      {props.title}
    </Button>
  );
}

const Style = StyleSheet.create({
  answerBtn: {
    marginHorizontal: 25,
  },
});

AnswerButton.propTypes = {
  title: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  callback: PropTypes.func.isRequired,
  answer: PropTypes.string,
  correct: PropTypes.bool,
};

export default AnswerButton;
