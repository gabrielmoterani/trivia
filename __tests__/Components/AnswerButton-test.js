import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import AnswerButton from '../../src/Components/AnswerButton';

const initialProps = {
  title: 'Test',
  value: 'test',
  callback: jest.fn(),
};

const setup = (props = initialProps) => shallow(<AnswerButton {...props} />);

describe('Test AnswerButton component', () => {
  it('should render properly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
