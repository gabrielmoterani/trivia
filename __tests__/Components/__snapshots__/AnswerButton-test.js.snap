// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AnswerButton component should render properly 1`] = `
<ButtonComponent
  appearance="outline"
  onPress={[Function]}
  size="large"
  status="primary"
  style={
    Object {
      "marginHorizontal": 25,
    }
  }
>
  Test
</ButtonComponent>
`;
